### PR TITLE
[image, FEM, LinearSolver] fix Eigen3 assertion with SVD

### PR DESCRIPTION
**Pull request by fredroy**
While investigating the error in  image/examples/distanceFieldFilter.scn (raised in #3329 for example),
Eigen3 was throwing an assertion in Debug (and surely throwing unexpected errors in Release)

https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/master/CI_CONFIG=ubuntu_clang,CI_PLUGINS=options,CI_TYPE=debug/3967/testReport/SceneTests/applications_plugins_image_examples/MeshToImageMultipleRoi_scn/
for an example of the dump on the CI

The problem was that if one wants to compute matrices ThinU & ThinV for Eigen::SVD, the given matrices need to be set with a dynamic size.
-> https://gitlab.com/libeigen/eigen/-/blob/3.4/Eigen/src/SVD/JacobiSVD.h#L640

So the solution are either:
 - compute full V and full U and keep fixed size matrices
 - or set the matrices dynamic and resize to 3x3

It appeared that this combination EigenSVD, ThinU/V and fixed-size matrices was used in PlasticMaterial (FEM.HyperElastic), TriangularFEMForceField (FEM.Elastic) and SVDLinearSolver (LinearSolver.Direct)

\+ the use of SVD in MeshToImage was a little weird (doing it on matrix filled of Zeros). It appears that the code fillling the matrix did disappear while the code was converted from NewMat to Eigen3, so I restored it.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[force-full-build][ci-build][with-all-tests]

**Comment by fredroy** [comment]
Why SofaPython3 does not compile in Debug: https://github.com/microsoft/onnxruntime/issues/9735
TLDR: would be fixed by updating pybind on the msvc CI (good luck to @hugtalbot 🎁)

**Comment by damienmarchal** [comment]
Beware merger's It is flagged ready/// but CI fails

**Comment by alxbilger** [comment]
@damienmarchal Only the debug builds fail, which was also the case before the PR. This PR fixes some of them.
